### DISCUSSION TITLE
How to enable new version of key-only formulae (specifically, `texinfo`)?

### DISCUSSION THREAD

**Discussion by Mick235711** [Homebrew/discussions#2208] [Writing Formulae/Casks] [answered]
I have been trying to write a new formula (`wget2`) for Homebrew Core, and find that it errored out when generating documentation and manpages in `make` step. The command failed with the following message:
```
./wget2.texi:3: warning: unrecognized encoding name `UTF-8'.
```
Searching for the issue, I found [gnuwget/wget2#469](https://gitlab.com/gnuwget/wget2/-/issues/469), and there it suggested that the `sed` and `texinfo` version on macOS is too outdated. I can solve the `sed` problem by using `depends_on "gnu-sed" => :build`, and add
``` ruby
ENV.prepend_path "PATH", Formula["gnu-sed"].libexec/"gnubin"
```
before calling `make`. However, when trying to deal with `texinfo`, I added `depends_on "texinfo" => :build` to the depends list (no need to change `PATH` since Homebrew environment favors key-only formulas), and it build without any problem. However, `brew audit` reported this as an error:
```
Dependency 'texinfo' is provided by macOS; please replace 'depends_on' with 'uses_from_macos'.
```
is there an attribute or something to specify that I need `depends_on` for newer version?

**Comment by SMillerDev** [comment, marked as answer]
Nope, just tell the maintainers that it needs a newer version and we can choose to ignore the CI message.

**Reply by Mick235711** [reply]
Thanks!
